test/test_cautodoc.py

```diff
@@ -54,7 +54,7 @@ def _sphinx_build(self, srcdir):
             output_filename = os.path.join(app.outdir, f'index.{output_suffix}')
 
         # Remove paths from warning output for comparison
-        output_errors = re.sub(rf'(?m)^{srcdir}/index.rst:[0-9]+: ([^:]*: )(/[a-zA-Z0-9._-]+)*/',
+        output_errors = re.sub(rf'(?m)^.*{srcdir}/index.rst:[0-9]+: ([^:]*: )(/[a-zA-Z0-9._-]+)*/',
```

Comment:
Do you think it may be worth checking where the prefix comes from? I don't think it matters all that much, but it still feels a bit weird. Maybe we could also have the regex match only what we know may appear instead of filtering away anything, including (e.g.) some root directory added by mistake.

Reply:
So I didn't dig into it that much, but apparently `/private` is a historical thing on macOS, and e.g. `/tmp` is a symlink to `/private/tmp`. The discrepancy might come from some parts normalizing the symlink away and some parts not.

I guess an alternative might be doing `os.path.abspath()` on the paths we get from `tempfile.TemporaryDirectory()` so everything is handled in the same "domain" so to speak.

Reply:
I see, matching anything including root paths may not be so weird given that `/private` is indeed a path and not some other kind of prefix. Otherwise, yours might be a better alternative to mine indeed if you do feel like changing it. I may be overlooking other complications though.

Reply:
This also fixes the issue, better?

```
diff --git a/test/test_cautodoc.py b/test/test_cautodoc.py
index 2c957d3d4b1f..178a6bf83290 100755
--- a/test/test_cautodoc.py
+++ b/test/test_cautodoc.py
@@ -63,12 +63,12 @@ class ExtensionTestcase(testenv.Testcase):
         with tempfile.TemporaryDirectory() as srcdir:
             with open(os.path.join(srcdir, 'index.rst'), 'w') as f:
                 f.write(input_str)
-            return self._sphinx_build(srcdir)
+            return self._sphinx_build(os.path.realpath(srcdir))
 
     def _sphinx_build_file(self, input_filename):
         with tempfile.TemporaryDirectory() as srcdir:
             shutil.copyfile(input_filename, os.path.join(srcdir, 'index.rst'))
-            return self._sphinx_build(srcdir)
+            return self._sphinx_build(os.path.realpath(srcdir))
 
     def get_output(self):
         input_str = ''.join([d.get_directive_string() for d in self.directives])

```

Reply:
abspath does *not* fix the issue, it's realpath that's needed.